pydantic/main.py

```diff
@@ -106,7 +101,11 @@ def __new__(
         Returns:
             type: the new class created by the metaclass
         """
-        if _base_class_defined:
+        # Note `ModelMetaclass` refers to `BaseModel`, but is also used to *create* `BaseModel`, so we need to keep
+        # track of whether we've created the `BaseModel` class yet, and therefore whether it's safe to refer to it. If
+        # it *hasn't* been created, we assume that the `__new__` call we're in the middle of is for the `BaseModel`
+        # class, since that's defined immediately after the metaclass.
+        if bases:
```

Comment:
It seems to me the comment above this line should be updated a bit. At least to say something like "`BaseModel` itself won't have any bases, but any subclass of it will"

Reply:
Updated the comment